### PR TITLE
Updated tests after last commit & debugged profitability of trader more

### PR DESCRIPTION
**Pull request by BensGitHub2022**
Updated account_sim and trade_executor_sim for improved functionality of trading in simulator mode to improve debugging profitability. 
Modified action_writer to include storing account balance and profit inside positions history.
Removed print account balance from trade_bot
Tested sim_trader on 1 years worth of data to check for different profitability conditions.

### CODE REVIEW

**Comment by BensGitHub2022** [comment]
Use this repo to generate large dataset if you want to test the code: https://github.com/BensGitHub2022/Data-Generator

**Comment by BensGitHub2022** [comment]
You need to adjust max_bars in your chart view in metatrader to get more than >100,000 candlesticks. The default is either 10,000 or 100,000 for most symbols - use this link to adjust your terminal settings: https://www.metatrader5.com/en/terminal/help/startworking/settings#max_bars

File: tests/context_mt5_test.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def test_connect_login_errors(self, mock_init, mock_login):
         with pytest.raises(PermissionError):
             self.context_with_complete_creds.connect()
 
+    # these tests no longer serve a purpose!
```
Comment (SaralynOgden) [inline tests/context_mt5_test.py]: Please remove them, if they are no longer necessary

Comment (SaralynOgden) [reply tests/context_mt5_test.py]: Actually, I disagree. This test is still relevant, it just need to be rewritten. [Here](https://github.com/BensGitHub2022/MT5-Expert-Advisor/blob/main/src/context_mt5.py#L10) in the code it is still possible to pass in a dictionary that doesn't have all of the needed values and we still need to check that this scenario is handled properly.

File: tests/context_mt5_test.py [hunk 2]
```diff
@@ -13,24 +13,24 @@ class ContextMT5Tests(unittest.TestCase):
     
     settings = json.loads("""{
```
Comment (SaralynOgden) [inline tests/context_mt5_test.py]: Since we are no longer using settings in the class tested, we should remove this dictionary.

Comment (BensGitHub2022) [reply tests/context_mt5_test.py]: Updated!

File: tests/context_mt5_test.py (hunk 1 above)
Comment (BensGitHub2022) [reply tests/context_mt5_test.py]: I think I understand what you mean, see update

File: tests/context_mt5_test.py [hunk 3]
```diff
@@ -52,13 +46,10 @@ def test_connect_login_errors(self, mock_init, mock_login):
         mock_login.return_value = False
         with pytest.raises(PermissionError):
             self.context_with_complete_creds.connect()
-
-    def test_connect_setting_value_missing(self):
-        context_with_no_settings = ContextMT5({}, self.credentials)
-        with pytest.raises(KeyError):
-            context_with_no_settings.connect()
     
+    # I think this is what you mean
```
Comment (SaralynOgden) [inline tests/context_mt5_test.py]: Yes. Please remove this comment before pushing :D
You can always leave comments like this on the PR so that you don't have to do another commit to remove them.

**Review by SaralynOgden — APPROVED** [review]

**Review by bpsix — APPROVED** [review]
